### PR TITLE
extend cell value load to support custom datetime format

### PR DESCRIPTION
**Pull request by artiz**
# PR Details

Add support for custom number formats to support all date-time variants parsing

## Description

We are using excelize in our ETL project tp parse variety XLSX files from our customers and had found that lots of required dates are parsed incorrecty as raw numbers. So we decided to extend dates support to use custom formats

## Related Issue
PR provides fix for https://github.com/360EntSecGroup-Skylar/excelize/issues/73

## Motivation and Context
PR provides support of custom Excel datetime formats

## How Has This Been Tested
New Unit Test added

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: templates.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ const templateContentTypes = `<Types xmlns="http://schemas.openxmlformats.org/pa
...
...org/markup-compatibility/2006" mc:Ignorable="x14ac x16r2" xmlns:x14ac="http://schemas.microsoft.com/office/spreadsheetml/2009/9/ac" xmlns:x16r2="http://schemas.microsoft.com/office/spreadsheetml/2015/02/main"><numFmts count="2"><numFmt formatCode="General" numFmtId="100"/><numFmt formatCode="[$-409]MM/DD/YYYY" numFmtId="101"/></numFmts><fonts count="1" x14ac:knownFonts="1"><font><sz val="11"/><color theme="1"/><name val="Calibri"/><family val="2"/></font></fonts><fills count="2"><fill><patternFill patternType="none"/></fill><fill><patternFill patternType="gray125"/></fill></fills><borders count="1"><border><left/><right/><top/><bottom/><diagonal/></border></borders><cellStyleXfs count="1"><xf numFmtId="0" fontId="0" fillId="0" borderId="0"/></cellStyleXfs><cellXfs count="3"><xf numFmtId="0" fontId="0" fillId="0" borderId="0" xfId="0"/><xf numFmtId="100" fontId="0" fillId="0" borderId="0" xfId="0"/><xf numFmtId="101" fontId="0" fillId="0" borderId="0" xfId="0"/></cellXfs><cellStyles count="1"><cellStyle name="Normal" xfId="0" builtinId="0"/></cellStyles><dxfs count="0"/><tableStyles count="0" defaultTableStyle="TableStyleMedium2" defaultPivotStyle="PivotStyleLight16"/></styleSheet>`
```
Comment (xuri) [inline templates.go:32]: That's unnecessary to add number format code 100 and 101 in the template.

File: go.mod [hunk 2]
```diff
@@ -1,4 +1,4 @@
-module github.com/360EntSecGroup-Skylar/excelize/v2
+module github.com/artiz/excelize
```
Comment (xuri) [inline go.mod:1]: the incorrect package name in the go module.

**Review by xuri — CHANGES_REQUESTED** [review]
Thanks for your PR, I have left some comments.

File: go.mod (hunk 2 above)
Comment (artiz) [reply go.mod:1]: yep, package with default name in go.mod https://github.com/artiz/excelize/tree/extended-parse-time could not be installed by `go mod` so it was temporary fix, will revert it

File: go.sum [hunk 3]
```diff
@@ -1,35 +1,50 @@
+github.com/creack/pty v1.1.9/go.mod h1:oKZEueFk5CKHvIhNR5MUki03XCEU+Q6VDXinZuGJ33E=
```
Comment (xuri) [inline go.sum:1]: Remove unused dependencies version checksum info

**Review by xuri — CHANGES_REQUESTED** [review]
In addition, I suggest removing personal info from the doc properties of the test spreadsheet files, and could you add some unit tests for the non covered codes?

File: templates.go (hunk 1 above)
Comment (artiz) [reply templates.go:32]: Yep, I did it to cover new logic that I had added, without it PR coverage check raises red flag.
https://github.com/artiz/excelize/commit/b1c578c0ba723285b6adba7eae9c6e9023f30322 - actually in this commit

**Comment by artiz** [comment]
> In addition, I suggest removing personal info from the doc properties of the test spreadsheet files, and could you add some unit tests for the non covered codes?

Got it, will do

**Review by xuri — CHANGES_REQUESTED** [review]
I have tested by the spreadsheet attachment `1_2017-06-29.xlsx` in the issue #73 uploaded by `vchugreev`, It seems like the value `29\ June\ 2017\ \г\.;@` after applying number format code contains escape characters,

File: go.sum (hunk 3 above)
Comment (artiz) [reply go.sum]: done

**Comment by artiz** [comment]
> I have tested by the spreadsheet attachment `1_2017-06-29.xlsx` in the issue #73 uploaded by `vchugreev`, It seems like the value `29\ June\ 2017\ \г\.;@` after applying number format code contains escape characters.

Fixed in https://github.com/artiz/excelize/commit/6e66d1dab42abf6bfce59685825e1ccd46bba2ad

**Comment by omaralsoudanii** [comment]
Hi @artiz @xuri 

Thank you for the amazing work you are doing, The library is pretty great when reading XLSX files, however I am facing the same problem when parsing dates. 

Some times the date value is parsed as string e.g: 25/03/2016, some times it's parsed as a float like 43345.5 and some times as an integer.

I believe this PR would probably solve it? if so do you have any estimate when this will be merged?

Thanks!

**Review by xuri — APPROVED** [review]

**Comment by xuri** [comment]
@artiz Thanks for your PR. @omaralsoudanii I have merged this PR.

**Pull request merged**